apps/site/lib/site_web/controllers/schedule/finder_api.ex

```diff
@@ -0,0 +1,370 @@
+defmodule SiteWeb.ScheduleController.FinderApi do
+  @moduledoc """
+    API for retrieving journeys for a route, and for
+    showing trip details for each journey on demand.
+
+    "Departure" here is analagous to PredictedSchedule.t()
+  """
+  use SiteWeb, :controller
+
+  alias Predictions.Prediction
+  alias Routes.Route
+  alias Schedules.{Schedule, Trip}
+  alias Site.TransitNearMe
+  alias SiteWeb.ScheduleController.TripInfo, as: Trips
+  alias SiteWeb.ScheduleController.VehicleLocations, as: Vehicles
+
+  import SiteWeb.ScheduleController.ScheduleApi, only: [format_time: 1, fares_for_service: 4]
+
+  @type react_keys :: :date | :direction | :is_current
+  @type react_strings :: [{react_keys, String.t()}]
+  @type converted_values :: {Date.t(), integer, boolean}
+
+  @type enhanced_journey :: %{
+          departure: PredictedSchedule.t() | nil,
+          arrival: PredictedSchedule.t() | nil,
+          trip: Trip.t() | nil,
+          realtime: TransitNearMe.time_data() | nil
+        }
+
+  # How many seconds a departure is considered recent
+  @recent_departure_retention_policy 600
+
+  # Leverage the JourneyList module to return a simplified set of trips
+  @spec journeys(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def journeys(conn, %{"stop" => stop_id, "date" => date} = params) do
+    {:ok, user_selected_date} = Date.from_iso8601(date)
+    {schedules, predictions} = load_from_repos(conn, params)
+
+    # Emulate original Schedules tab journey build logic:
+    today? = params["is_current"] == "true"
+    current_time = if today?, do: user_selected_date, else: nil
+
+    journey_list_opts = [
+      origin_id: stop_id,
+      current_time: current_time
+    ]
+
+    journeys =
+      schedules
+      |> JourneyList.build(predictions, :predictions_then_schedules, true, journey_list_opts)
+      |> prepare_journeys_for_json()
+
+    json(conn, journeys)
+  end
+
+  # Use alternative JourneyList constructor to only return trips with predictions
+  @spec departures(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def departures(conn, %{"stop" => stop_id} = params) do
+    now = conn.assigns.date_time |> DateTime.to_date() |> Date.to_iso8601()
+    params = %{"date" => now, "is_current" => "true"} |> Map.merge(params)
+
+    {schedules, predictions} = load_from_repos(conn, params)
+
+    journeys =
+      schedules
+      |> JourneyList.build_predictions_only(predictions, stop_id, nil)
+      |> Map.get(:journeys, [])
+      |> Enum.map(&enhance_journeys/1)
+      |> prepare_journeys_for_json()
+
+    json(conn, journeys)
+  end
+
+  # Return a modified %TripInfo{} map by building up the conn.
+  # Simulates the :trip_info data generated by the schedules/x/schedule tab
```

Comment:
As above: why?

Reply:
We wanted to avoid modifying the existing architecture which is still used by Bus. The existing TripInfo architecture relies on the `conn` and assigns all data into that. The `trip_info` assign also has a long list of pre-requisites (things that also need to be assigned in the conn), which cannot easily be accessed directly. Hence the long pipeline of conn operations leading up to getting the `trip_info` data in the Finder API.